scapy/contrib/automotive/bmw/doip.py

```diff
@@ -45,8 +45,7 @@ def extract_padding(self, s):
         return s[:8], s[8:]
 
 
-bind_layers(TCP, DoIP, sport=6801)
-bind_layers(TCP, DoIP, dport=6801)
+bind_layers(TCP, DoIP, sport=6801, dport=6801)
```

Comment:
You should do
```
 bind_bottom_up(TCP, DoIP, sport=6801)
 bind_bottom_up(TCP, DoIP, dport=6801)
 bind_layers(TCP, DoIP, sport=6801, dport=6801)
```

So that when dissecting, any of ports triggers DoIP, and that both are set when building

Reply:
Thanks for pointing this out.
You're right, I would have changed the behavior.

Can the last `bind_layers` call be replaced by a `bind_top_down` call? If I'm thinking correct, the `bind_layers` call would add one more `bottom up` relation with dport and sport connected with an `and`, while these two properties are already in the same relation with an `or`.

Reply:
I think the bind_layers is enough. It’s ok to have he double relation.

It does not really matter, it’s up to you

Reply:
You need to keep the `bind_layers(TCP, DoIP, sport=6801)` (and same with `dport`) to make sure that packets with source port **or** destination port 6801 get dissected as `DoIP`.

Reply:
In this way (dissection), `bind_bottom_up` is enough (but your comment stands true otherwise)

Reply:
You're right.